skopt/tests/test_searchcv.py

```diff
@@ -124,3 +124,102 @@ def test_searchcv_runs_multiple_subspaces():
     # test if all subspaces are explored
     total_evaluations = len(opt.cv_results_['mean_test_score'])
     assert total_evaluations == 1+1+2, "Not all spaces were explored!"
+
+
+def test_searchcv_sklearn_compatibility():
+    """
+    Test whether the BayesSearchCV is compatible with base sklearn methods
+    such as clone, set_params, get_params.
+    """
+
+    X, y = load_iris(True)
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, train_size=0.75, random_state=0
+    )
+
+    # used to try different model classes
+    pipe = Pipeline([
+        ('model', SVC())
+    ])
+
+    # single categorical value of 'model' parameter sets the model class
+    lin_search = {
+        'model': Categorical([LinearSVC()]),
+        'model__C': Real(1e-6, 1e+6, prior='log-uniform'),
+    }
+
+    dtc_search = {
+        'model': Categorical([DecisionTreeClassifier()]),
+        'model__max_depth': Integer(1, 32),
+        'model__min_samples_split': Real(1e-3, 1.0, prior='log-uniform'),
+    }
+
+    svc_search = {
+        'model': Categorical([SVC()]),
+        'model__C': Real(1e-6, 1e+6, prior='log-uniform'),
+        'model__gamma': Real(1e-6, 1e+1, prior='log-uniform'),
+        'model__degree': Integer(1, 8),
+        'model__kernel': Categorical(['linear', 'poly', 'rbf']),
+    }
+
+    opt = BayesSearchCV(
+        pipe,
+        [(lin_search, 1), svc_search],
+        n_iter=2
+    )
+
+    opt_clone = clone(opt)
+
+    params, params_clone = opt.get_params(), opt_clone.get_params()
+    assert params.keys() == params_clone.keys()
+
+    for param, param_clone in zip(params.items(), params_clone.items()):
+        assert param[0] == param_clone[0]
+        assert isinstance(param[1], type(param_clone[1]))
+
+    opt.set_params(search_spaces=[(dtc_search, 1)])
+
+    opt.fit(X_train, y_train)
+    opt_clone.fit(X_train, y_train)
+
+    total_evaluations = len(opt.cv_results_['mean_test_score'])
+    total_evaluations_clone = len(opt_clone.cv_results_['mean_test_score'])
+
+    # test if expected number of subspaces is explored
+    assert total_evaluations == 1
+    assert total_evaluations_clone == 1+2
+
+
+def test_searchcv_reproducibility():
+    """
+    Test whether results of BayesSearchCV can be reproduced with a fixed
+    random state.
+    """
+
+    X, y = load_iris(True)
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, train_size=0.75, random_state=0
+    )
+
+    random_state = 42
+
+    opt = BayesSearchCV(
+        SVC(random_state=random_state),
+        {
+            'C': Real(1e-6, 1e+6, prior='log-uniform'),
+            'gamma': Real(1e-6, 1e+1, prior='log-uniform'),
+            'degree': Integer(1, 8),
+            'kernel': Categorical(['linear', 'poly', 'rbf']),
+        },
+        n_iter=11, random_state=random_state
+    )
+
+    opt.fit(X_train, y_train)
+    best_estimator = opt.best_estimator_
+
+    opt2 = clone(opt).fit(X_train, y_train)
```

Comment:
I wonder if there might be a situation where the test will pass because the estimator being cloned was already fitted? That is, when the estimator is cloned before calling `fit` method, the test would fail. From looking at definition of clone method, I am not sure. If you think this might happen, consider adding another copy before fitting, and comparing it with the rest.

Reply:
I don't think this is an issue, I think clone always return a new un-fitted copy of the estimator, regardless of wether the input estimator is fitted or not. I'll double check it.

Reply:
Indeed, you can check https://github.com/scikit-learn/scikit-learn/blob/19eb62ff/sklearn/base.py#L30 and https://github.com/scikit-learn/scikit-learn/blob/19eb62ff/sklearn/base.py#L198 : `clone` return a fresh instance of the same class, and get the parameters from the signature of the `__init__` of the cloned instance.